test/jdk/java/nio/file/Path/PathOps.java

```diff
@@ -1437,6 +1437,30 @@ static void doWindowsTests() {
         int h2 = test("c:\\FOO").path().hashCode();
         if (h1 != h2)
             throw new RuntimeException("PathOps failed");
+
+        // long path prefixes
+        test("\\\\?\\C:\\mnt\\file.dat")  // absolute
+            .string("C:\\mnt\\file.dat");
+        test("\\\\?\\\\\\server\\share\\dir\\file.dat")  // UNC
+            .invalid();
+        test("\\\\?\\file.dat")           // relative
+            .invalid();
+        test("\\\\?\\\\file.dat")         // directory-relative
+            .invalid();
+        test("\\\\?\\C:file.dat")         // drive-relative
+            .invalid();
+        test("\\\\?\\")                   // empty
+            .invalid();
+
+        // long UNC path prefixes
+        test("\\\\?\\UNC\\server\\share\\dir\\file.dat")      // UNC
+            .string("\\\\server\\share\\dir\\file.dat");
+        test("\\\\?\\UNC\\server\\share\\C:\\mnt\\file.dat")  // absolute
```

Comment:
Shouldn't this be "\\\\?\\UNC\\C:\\file.dat"?

Also can we try a relative path, e.g.  "\\\\?\\UNC\\file.dat"

Reply:
Updated and amended in 90ef2d30a9ed8f7ddb5891836663e5f8728a948d.